src/dbus_api/types.rs

```diff
@@ -119,3 +118,33 @@ impl DataType for TData {
     type Method = ();
     type Signal = ();
 }
+
+/// An action queue.
+/// Add and remove actions are pushed onto the queue.
+/// The queue can also be drained.
+#[derive(Debug)]
+pub struct ActionQueue {
+    queue: VecDeque<DeferredAction>,
+}
+
+impl ActionQueue {
```

Comment:
I'm happy to leave the ActionQueue commit on the head of PR #193 instead of on the tail of this PR. I expect to be able to make it more valuable, but not until some other work is done.